proposals/20200504-falco-artifacts-scope.md

```diff
@@ -0,0 +1,191 @@
+
+**package**
+
+*An installable artifact that is operating system specific. All packages MUST be hosted on bintray for X86_64.*
+
+**image**
+
+*OCI compliant container image hosted on dockerhub with tags for every release.*
+ 
+
+# Packages
+
+Official packages for X86_64 only. All package names should contain a version which is not specified below.
+
+The following convention MUST be used for all packages.
+
+_If a package installs Falco as a daemon (systemd, init) it MUST contain `daemon`._
+
+_If a package installs the Falco kernel module it MUST contain `module`._
+
+_If a package installs the Falco BPF probe it MUST contain `bpf`._
+
+_If a package installs the Falco kernel module it MUST contain `module`._
+
+---
+
+### .deb
+
+ Falco running in debian like systems that will default to the kernel module.
+
+- falco-x86.deb
+     - `falco` only (default depends on `falco-module`)
+ - falco-daemon-x86.deb
+     - `falco` and script to configure as daemon (default depends on `falco-module`)
+ - falco-module-x86.deb
+     - `module` only
+ - falco-bpf-x86.deb
+     - `bpf` only
+ - falco-ptrace-x86.deb
+     - `ptrace` only
+
+### .rpm
+
+ Falco running in rpm like systems that will default to the kernel module.
+
+- falco-x86.deb
+     - `falco` only (default depends on `falco-module`)
+ - falco-daemon-x86.deb
+     - `falco` and script to configure as daemon (default depends on `falco-module`)
+ - falco-module-x86.deb
+     - `module` only
+ - falco-bpf-x86.deb
+     - `bpf` only
+ - falco-ptrace-x86.deb
+     - `ptrace` only
+
+
+### .tar.gz
+
+ - falco-src-x86.tar.gz
+     - No binaries
+     - `INSTALL` file
+ - falco-driver-src-x86.tar.gz
+     - `falco` and `/driver` with `Makefile`
+     - `INSTALL` file
+ - falco-driver-download-x86.tar.gz
+     - `falco` and `falco-driver-loader` download only
+ - falco-driver-full-x86.tar.gz
+     - `falco` and `falco-driver-loader` download else compile
+
+
+# Images
+
+The following convention MUST be used for all container images.
+
+_If a container image requires `--privileged` the name MUST have the `priviliged-` prefix._
+
+_If a container image is for development purproses it MUST have the `dev-` prefix._ 
+
+_If a container image is used to install something, then exit it MUST have the `-install` suffix.*_
+
+---
+
+ - falcosecurity/falco:TAG
+     - Runs `falco` userspace only
+ - falcosecurity/privileged-driver-install:TAG
+     - Runs `falco-driver-loader` and exit
+ - falcosecurity/priviliged-driver-install-falco 
+     - first runs `privileged-driver-install` then runs `falco`
+ - falcosecurity/priviliged-host-systemd-driver-falco:TAG
+     - Detects arch and runs new bash script:
+         - Installs Falco with packages (deb, rpm, src)
+         - Configures with systemd and Unix Domain Socket [#858](https://github.com/falcosecurity/falco/issues/858)
```

Comment:
During past community calls and discussions, we all agreed on reducing the artifacts (and their scope) supported in the main Falco repo. Or, at least, this was what I understood.

I would keep the `systemd` unit definition here. But the process of installing it via a docker image that `nsenter` into the host is a task for `falcoctl` (or even `contrib`), imho.

So, I think this installation pattern is better kept in another place rather than in the main repository.